cylc/flow/network/ssh_client.py

```diff
@@ -45,10 +46,33 @@ def __init__(
             timeout: Union[float, str] = None
     ):
         self.workflow = workflow
-
+        self.SLEEP_INTERVAL = 0.1
         if not host:
             self.host, _, _ = get_location(workflow)
 
+    async def async_request(self, command, args=None, timeout=None):
+        """Send asynchronous request via SSH.
+        """
+        cmd, ssh_cmd, login_shell, cylc_path, message = self.prepare_command(
+            command, args, timeout)
+        proc = _remote_cylc_cmd(
+            cmd,
+            host=self.host,
+            stdin_str=message,
+            ssh_cmd=ssh_cmd,
+            remote_cylc_path=cylc_path,
+            ssh_login_shell=login_shell,
+            capture_process=True)
+        while True:
+            if proc.poll() is not None:
+                break
+            await asyncio.sleep(self.SLEEP_INTERVAL)
```

Comment:
Ooh, just spotted we hadn't implemented `timeout` for this client, would need to wrap with `async_timeout` or compare `time() - time_started`, can bump that though.

Reply:
So we have added the timeout in this client to the command (in `prepare_command`), via the comms_timeout cli option. Do you think that will need changing?

Reply:
That passes the timeout through to the ZMQ client, but I don't think it protects us from network issues between the hosts on either end of the SSH command.